src/fileio.cpp

```diff
@@ -16,16 +16,15 @@
 #include "fileio.h"
 #include "rev.h"
 #ifdef LINUX
-	#include "unix/fileio_unix.h"
 	#include <dirent.h>
 	#include <errno.h>
 	#include <unistd.h>
 #elif WINDOWS
-	#include "windows/fileio_windows.h"
 	#include <direct.h> // contains chdir in windows
 #endif
 #include <sys/types.h>
-#include <sys/stat.h>
+
+const char DIR_SEP = std::filesystem::path::preferred_separator;  ///< Directory separator character.
```

Comment:
```suggestion
constexpr auto DIR_SEP = std::filesystem::path::preferred_separator;
```
I think this comment is superfluous now that we're using a well-documented built-in property of the standard library. Also, using `constexpr` should be fine here. Additionally, I'm not sure about putting `char` here. `std::filesystem::path::preferred_separator` has the type `wchar_t` on Windows. Maybe `auto` makes sense?